google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/st/STBigQueryStorageLongRunningWriteTest.java

```diff
@@ -0,0 +1,245 @@
+      Logger.getLogger(ITBigQueryStorageLongRunningTest.class.getName());
+
+  private static final String LONG_TESTS_ENABLED_PROPERTY =
+      "bigquery.storage.enable_long_running_tests";
+
+  private static final String LONG_TESTS_DISABLED_MESSAGE =
+      String.format(
+          "BigQuery Storage long running tests are not enabled and will be skipped. "
+              + "To enable them, set system property '%s' to true.",
+          LONG_TESTS_ENABLED_PROPERTY);
+  private static BigQueryWriteClient client;
+  private static String parentProjectId;
+  private static final String DATASET = RemoteBigQueryHelper.generateDatasetName();
+  private static final String TABLE = "testtable";
+  private static final String TABLE2 = "testtable2";
+  private static final String DESCRIPTION = "BigQuery Write Java long test dataset";
+
+  private static TableInfo tableInfo;
+  private static TableInfo tableInfo2;
+  private static String tableId;
+  private static String tableId2;
+  private static BigQuery bigquery;
+
+  private static JSONObject MakeJsonObject(int size) throws IOException {
+    JSONObject object = new JSONObject();
+    // size: (1, simple)(2,complex)()
+    if (size == 1) {
+      object.put("test_str", "aaa");
+      object.put("test_numerics", new JSONArray(new String[]{"1234", "-900000"}));
+      object.put("test_datetime", String.valueOf(LocalDateTime.now()));
+    }
+    else if (size == 2) {  // make it complicated and large
+      // TODO(jstocklass): Make a better json object that doesn't break the format rules.
+    }
+    return object;
+  }
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    Assume.assumeTrue(LONG_TESTS_DISABLED_MESSAGE, Boolean.getBoolean(LONG_TESTS_ENABLED_PROPERTY));
+    client = BigQueryWriteClient.create();
+    parentProjectId = String.format("projects/%s", ServiceOptions.getDefaultProjectId());
+    RemoteBigQueryHelper bigqueryHelper = RemoteBigQueryHelper.create();
+    bigquery = bigqueryHelper.getOptions().getService();
+    DatasetInfo datasetInfo = DatasetInfo.newBuilder(/* datasetId = */ DATASET).setDescription(DESCRIPTION).build();
+    bigquery.create(datasetInfo);
+    LOG.info("Created test dataset: " + DATASET);
+    tableInfo =
+        TableInfo.newBuilder(
+            TableId.of(DATASET, TABLE),
+            StandardTableDefinition.of(
+                Schema.of(
+                    com.google.cloud.bigquery.Field.newBuilder("foo", LegacySQLTypeName.STRING)
+                        .setMode(Field.Mode.NULLABLE)
+                        .build())))
+            .build();
+    com.google.cloud.bigquery.Field.Builder innerTypeFieldBuilder =
+        com.google.cloud.bigquery.Field.newBuilder(
+            "inner_type",
+            LegacySQLTypeName.RECORD,
+            com.google.cloud.bigquery.Field.newBuilder("value", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.REPEATED)
+                .build());
+
+    tableInfo2 =
```

Comment:
Remove if you are not working on it yet. Make a PR single purpose and easier for review.

Reply:
Yeah I'll remove other unused things as well